src/lsp-client/LSPReaderThread.cpp

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023, Andrea Anzani <andrea.anzani@gmail.com>
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+
+
+#include "LSPReaderThread.h"
+#include "LSPPipeClient.h"
+
+LSPReaderThread::LSPReaderThread(LSPPipeClient& looper):
+	GenericThread("LSPReaderThread"), fTransport(looper)
+{
+}
+
+status_t
+LSPReaderThread::ExecuteUnit(void)
+{
+	if(fTransport.readStep() == false){
+		printf("LSPReaderThread::ExecuteUnit(void) false\n");
+		Quit();
+		fTransport.Close();  //FIX ME: this should be performed by the Client itself not from the thread!
```

Comment:
Do we still need to fix this?

Reply:
Not really.. Would be nicer to slithly refactor this code but at this point maybe is better to move directly to the new Task class